### PR TITLE
Support curated return value for network interface and refine test

### PR DESCRIPTION
**Pull request by yuwzho**
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

azure_rm_networkinterface

##### ADDITIONAL INFORMATION

```paste below
2.7.0
```

### CODE REVIEW

**Comment by Fred-sun** [comment]
@zikalino @yungezz

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py [hunk 1]
```diff
@@ -104,6 +104,97 @@
...
+                    description:
+                        - Whether the ip configuration is the primary one in the list.
+        enable_accelerated_networking:
+            description:
+                - Specifies whether the network interface should be created with the accelerated networking feature or not
+        create_with_security_group:
+            description:
+                - Specifies whether a default security group should be be created with the NIC. Only applies when creating a new NIC.
+            type: bool
+            version_added: 2.6
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py]: removed?

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py [hunk 2]
```diff
@@ -150,57 +282,50 @@ def exec_module(self, **kwargs):
...
+            results = self.get_item()
         elif self.resource_group:
-            self.results['ansible_facts']['azure_networkinterfaces'] = self.list_resource_group()
+            results = self.list_resource_group()
         else:
-            self.results['ansible_facts']['azure_networkinterfaces'] = self.list_all()
+            results = self.list_all()
 
+        self.results['ansible_facts']['azure_networkinterfaces'] = self.serialize_nics(results)
+        self.results['ansible_facts']['azure_rm_networkinterfaces'] = self.to_dict_list(results)
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py]: remove ansible_facts and  networkinterfaces is enough

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py [hunk 3]
```diff
@@ -104,6 +104,95 @@
         "tags": {},
         "type": "Microsoft.Network/networkInterfaces"
     }]
+azure_rm_networkinterfaces:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py]: networkinterfaces?

File: lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py [hunk 4]
```diff
@@ -150,57 +282,50 @@ def exec_module(self, **kwargs):
...
-            self.results['ansible_facts']['azure_networkinterfaces'] = self.get_item()
+            results = self.get_item()
         elif self.resource_group:
-            self.results['ansible_facts']['azure_networkinterfaces'] = self.list_resource_group()
+            results = self.list_resource_group()
         else:
-            self.results['ansible_facts']['azure_networkinterfaces'] = self.list_all()
+            results = self.list_all()
 
+        self.results['ansible_facts']['azure_networkinterfaces'] = self.serialize_nics(results)
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_networkinterface_facts.py:294]: @zikalino  should we keep this in output? or keep both format, which makes the output body big

**Review by yungezz — COMMENTED** [review]
LGTM.  some minor change need.

**Review by yungezz — APPROVED** [review]

**Pull request merged**